raiden/tests/utils/transport.py

```diff
@@ -145,67 +139,76 @@ def __getattr__(self, item):
 class AdminUserAuthProvider:
     __version__ = "0.1"
 
-    def __init__(self, config, account_handler):
+    def __init__(self, config, account_handler) -> None:  # type: ignore
         self.account_handler = account_handler
         self.log = logging.getLogger(__name__)
-        self.credentials = config["admin_credentials"]
+        if "credentials_file" in config:
+            credentials_file = Path(config["credentials_file"])
+            if not credentials_file.exists():
+                raise AssertionError(f"Credentials file '{credentials_file}' is missing.")
+            try:
+                self.credentials = json.loads(credentials_file.read_text())
+            except (JSONDecodeError, UnicodeDecodeError, OSError) as ex:
+                raise AssertionError(
+                    f"Could not read credentials file '{credentials_file}': {ex}"
+                ) from ex
+        elif "admin_credentials" in config:
+            self.credentials = config["admin_credentials"]
+        else:
+            raise AssertionError(
+                "Either 'credentials_file' or 'admin_credentials' must be specified in "
+                "auth provider config."
+            )
```

Comment:
Do we really need all these cases, now?

Reply:
This now covers both cases, the one used in the tests as well as the RSB setup.
The plan is to have this code only once and not two slightly different versions in here and in the RSB.